heartbeat/monitors/active/http/task.go

```diff
@@ -198,6 +202,9 @@ func buildRequest(addr string, config *Config, enc contentEncoder) (*http.Reques
 
 		request.Header.Add(k, v)
 	}
+	if ua := request.Header.Get("User-Agent"); ua == "" {
+		request.Header.Set("User-Agent", userAgent)
+	}
 
 	if enc != nil {
 		enc.AddHeaders(&request.Header)
```

Comment:
concerning the implementation, you might want to take a look at https://github.com/elastic/beats/blob/fleet/agent/kibana/round_trippers.go#L33-L52 and use http.Roundtripper interface.

Reply:
@andrewvc ^

Reply:
What's the advantage of adding it there vs. here? Seems like they're fairly equivalent in this case.

Reply:
There are, but you make sure that all connection through the http client get the headers no matter what.

Reply:
But, you are right I am not familliar enough with hearthbeat client. so it might be equivalent in that case.

Reply:
I think in our case it's equivalent since this is the only entry point through to the client. I think I'll leave the code here so there's only one spot where header stuff happens, which at least to me, seems easier to maintain, but I agree with you, generally speaking if you share the HTTP client it should go in the round tripper.